### PR TITLE
cleanup of signed/unsigned vars and const usage

### PR DESCRIPTION
**Pull request by laroche**
Try to avoid mixing signed/unsigned vars and also be more strict with "const" usage.

### CODE REVIEW

**Comment by laroche** [comment]
I've re-run the tests locally and they finish ok. Is this some race conditions within the tests?
(I'm rather new to zephyr.)

best regards,

Florian La Roche

File: lib/libc/minimal/source/string/strerror.c [hunk 1]
```diff
@@ -28,7 +28,7 @@ char *strerror(int errnum)
 		return (char *)sys_errlist[errnum];
 	}
 
-	return "";
+	return (char *)"";
```
Comment (keith-packard) [inline lib/libc/minimal/source/string/strerror.c:31]: yup, required to cast from (const char *).

**Review by keith-packard — COMMENTED** [review]
I don't know about the zephyr-specific changes, but the change to minimal libc looks correct.

File: arch/arm64/core/mmu.c [hunk 2]
```diff
@@ -997,7 +997,7 @@ size_t arch_virt_region_align(uintptr_t phys, size_t size)
 {
 	size_t alignment = CONFIG_MMU_PAGE_SIZE;
 	size_t level_size;
-	int level;
+	unsigned int level;
```
Comment (peter-mitsis) [inline arch/arm64/core/mmu.c:1000]: I don't think that this is safe as it looks like BASE_XLAT_LEVEL can be 0. The following loop might never end in that case.

File: kernel/smp.c [hunk 3]
```diff
@@ -115,7 +115,7 @@ void z_smp_init(void)
 
 	unsigned int num_cpus = arch_num_cpus();
 
-	for (int i = 1; i < num_cpus; i++) {
+	for (unsigned int i = 1; i < num_cpus; i++) {
```
Comment (peter-mitsis) [inline kernel/smp.c:118]: Observation: Although this fixes the mixing of unsigned and int variables on this line, it introduces subsequent mixing a few lines down with start_cpu().

**Comment by npitre** [comment]
The change to arch/arm64/core/mmu.c` is wrong.

**Comment by laroche** [comment]
I'll review over the weekend to check things again.

**Pull request closed without merging**